content/docs/command-reference/remote/modify.md

```diff
@@ -333,14 +333,21 @@ following parameters (listed in order or precedence) to customize the
 authentication method:
 
 1. `connection_string` is used for authentication if given (all others params
-   are ignored).
+   are ignored). If `connection_string` is not set, the `account_name` must be
+   given.
 2. If `tenant_id` and `client_id` or `client_secret` are given, Active Directory
    (AD)
    [service principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
    auth is performed.
-3. The storage `account_name` is tried next, along with `account_key` or
-   `sas_token` (in that order). If neither key nor token is provided, DVC will
-   try to connect anonymously.
+3. If either `account_key` or `sas_token` (in this order) are set, then DVC will
+   try to connect with those options.
+4. If `allow_anonymous_login` is set to `True`, then DVC will try to connect
+   anonymously.
```

Comment:
Unlike the other things in this list, this doesn't appear to be a "step." It's just a config flag. Maybe move to a note below this list?

Reply:
Ah these aren't steps @casperdcl, the list reflects the order in which auth params are used by DVC (Azure remotes). So if neither 1, 2, or 3 are set, this one is checked. I'd just add a reference to what anonymously is:

```suggestion
4. If `allow_anonymous_login` is set to `True`, then DVC will try to connect
   [anonymously]((https://docs.microsoft.com/en-us/azure/storage/blobs/anonymous-read-access-configure)).
```

Reply:
> the list reflects the order in which auth params are used

exactly.

- `allow_anonymous_login` is not in my mind an "auth param" but rather an "auth modifier option" so is out of place in this list
- is it really checked only at the fourth step (i.e. ignored if steps 1-3 succeed)?

The list may be correct (I haven't reviewed the underlying code to check) but I doubt it.

Reply:
> is it really checked only at the fourth step (i.e. ignored if steps 1-3 succeed)?

yes.

Reply:
> allow_anonymous_login is not in my mind an "auth param"

I see what you meant. We call all of these (config) params. They're params that can be sent `remote modify` (for Azure remotes)